### PR TITLE
feat: implement page copy between personal and note contexts (issue #713 Phase 3)

### PR DESCRIPTION
**Pull request by otomatty**
## 概要

ノートと個人ページ間でのページコピー機能を実装します。個人ページをノートにコピーしたり、ノートページを個人にコピーしたりできるようになります。元のページは変更されず、新しいコピーが作成されます。Issue #713 Phase 3 の実装です。

### バックエンド API
- `POST /api/notes/:noteId/pages/copy-from-personal/:pageId` エンドポイント追加
  - 個人ページ（`noteId IS NULL`）をノートネイティブページとしてコピー
  - 元ページは個人 /home に残り、コピーのみがノートに出現
  - `sourcePageId` で出自を記録
  - ページ内容（Y.js ドキュメント）も同時にコピー

- `POST /api/notes/:noteId/pages/:pageId/copy-to-personal` エンドポイント追加
  - ノートネイティブページを個人ページとしてコピー
  - 元ページはノートに残り、コピーのみが個人 /home に出現
  - ノート脱退後もコピーは独立して存在
  - 任意のロール（owner/member/guest）でコピー可能

### フロントエンド
- `NotePageView` にドロップダウンメニューを追加
  - 「個人に取り込み」アクション実装
  - コピー成功時にトーストで新ページへのリンク提供

- `NoteAddPageDialog` に「コピー」ボタン追加
  - 個人ページをノートにコピーするオプション
  - 既存の「リンク」機能と並行して提供

- API クライアント (`apiClient.ts`) に新メソッド追加
  - `copyPersonalPageToNote()`
  - `copyNotePageToPersonal()`

- React Query フック追加
  - `useCopyPersonalPageToNote()`
  - `useCopyNotePageToPersonal()`

### テスト
- バックエンド: 8 つの新しいテストケース
  - 正常系（コンテンツあり/なし）
  - 権限チェック（所有者確認、編集権限確認）
  - バリデーション（個人ページ確認、ノート所属確認）

- フロントエンド: API クライアントテスト追加
  - URL エンコーディング検証
  - レスポンス形式検証

### 国際化
- 日本語・英語の翻訳キー追加
  - `pageCopiedToNote`, `pageCopyToNoteFailed`
  - `pageCopiedToPersonal`, `pageCopyToPersonalFailed`
  - `copyToNote`, `copyToPersonal`
  - `open`, `moreActions`

## 変更の種類

- [x] ✨ 新機能 (New feature)
- [x] 🧪 テスト (Tests)

## テスト方法

1. バックエンドテストが全てパス: `npm test -- pages.test.ts`
2. フロントエンドテストが全てパス: `npm test -- apiClient.test.ts`
3. UI で個人ページをノートにコピー可能か確認
4. UI でノートページを個人にコピー可能か確認
5. コピー後、元ページと新ページが独立していることを確認

## チ

https://claude.ai/code/session_0176Qh5Q8Bt4Mrxset8sSk5j

---

<a href="https://app.devin.ai/review/otomatty/zedi/pull/716" target="_blank">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://static.devin.ai/assets/gh-open-in-devin-review-dark.svg?v=1">
    Open in Devin Review
  </picture>
</a>

## Summary by CodeRabbit

* **New Features**
  * Add two page copy flows: personal→note and note-native→personal; UI shows “Copy to Personal” only for note-native pages, toasts confirm success and may include an “Open” action when the copy is imported locally.
  * Note listings now indicate page scope (note-native vs linked personal) so UI can surface scope-specific actions.

* **Localization**
  * Added English and Japanese strings for copy/link actions and success/failure messages.

* **Tests**
  * Extensive tests covering copy flows, edge cases, permissions, and client behavior.

### CODE REVIEW

File: src/pages/NotePageView.tsx [hunk 1]
```diff
@@ -269,6 +280,35 @@ const NotePageView: React.FC = () => {
...
+        ),
+      });
+    } catch (error) {
+      console.error("Failed to copy note page to personal:", error);
+      toast({
+        title: t("notes.pageCopyToPersonalFailed"),
+        variant: "destructive",
+      });
+    }
+  };
```
[the remark this answers is not in the document]
Comment (otomatty) [reply src/pages/NotePageView.tsx:319]: ご指摘ありがとうございます。`useCallback` は意図的に外していて、戻すと lint が止まってしまうので、このままにさせてください。

- 最初の実装では `const handleCopyToPersonal = useCallback(async () => { ... }, [copyToPersonalMutation, navigate, noteId, page?.id, t, toast])` で書いたのですが、プロジェクトの `eslint` が `Compilation Skipped: Existing memoization could not be preserved` を **error** で返して lint が red に落ちました（`react-hooks` の React Compiler ルール経由）。
- このリポジトリでは React Compiler がメモ化を自動でかけてくれる前提なので、無闇に `useCallback` を入れるとコンパイラが既存メモ化を巻き取れずバックオフし、むしろ最適化を潰す形になります（参考: React Compiler の [“existing memoization preserved”](https://react.dev/learn/react-compiler#existing-memoization) ガイドライン）。同じファイル内の `handleBack` 等は依存が安定なので useCallback でも通りますが、`copyToPersonalMutation`（`useMutation` が返すオブジェクトは状態遷移ごとに identity が変わる）や `page?.id` のような optional-chain を含むと compiler がメモ化保存不能と判定します。
- 実際の再レンダリング負荷としても、`handleCopyToPersonal` はドロップダウンの `onClick` にだけ渡っており、メモ化差分が効く範囲は Radix の `DropdownMenuItem` 内に閉じている（しかもドロップダウンは閉じている間はアンマウント寄り）ので、コンパイラ自動メモ化で十分ワークロードはカバーできます。

というわけで現状の「useCallback なし」で lint と compiler の両方を通したい、というのが意図でした。もし React Compiler を前提にしない別運用に変えたい場合は教えてください、その場合は手動メモ化に統一して lint ルールごと調整する方向で動けます。

---
_Generated by [Claude Code](https://claude.ai/code)_
